src/compas_model/model/model.py

```diff
@@ -345,6 +350,16 @@ def add_interaction(self, a, b, interaction=None):
         edge = self._graph.add_edge(node_a, node_b, interaction=interaction)
         return edge
 
+    def add_material(self, material):
+        """Add a material to the model.
+
+        Parameters
+        ----------
+        material : :class:`Material`
+            The material to add.
+        """
+        self._materials.append(material)
```

Comment:
perhaps it would be useful to write out a use case with materials to see if you will want to look up materials using their key or though another mechanism. if key, then i would make the material store a dict...